modules/network/src/main/java/org/apache/ignite/internal/network/serialization/ClassDescriptor.java

```diff
@@ -172,4 +172,50 @@ public boolean hasWriteReplace() {
     public boolean hasReadResolve() {
         return serialization.hasReadResolve();
     }
+
+    /**
+     * Returns {@code true} if this is the descriptor of {@code null} values.
+     *
+     * @return {@code true} if this is the descriptor of {@code null} values
+     */
+    public boolean isNull() {
+        return clazz == Null.class;
+    }
+
+    /**
+     * Returns {@code true} if this is the descriptor of {@link java.util.Collections#singletonList(Object)} type.
+     *
+     * @return {@code true} if this is the descriptor of {@link java.util.Collections#singletonList(Object)} type
+     */
+    public boolean isSingletonList() {
+        return descriptorId == BuiltinType.SINGLETON_LIST.descriptorId();
```

Comment:
why is it different from `isNull` in the means of comparison?

Reply:
Thanks, it really makes sense to have more consistency here. Fixed.